Fixed #15009: Scroll to the top while navigating to the static pages.

ShivamJhaa opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #15009 
2. This PR does the following: Scroll to the top while navigating to the static pages.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct
Before:

https://user-images.githubusercontent.com/86483059/155546593-265e2875-5d46-476e-aee5-e56982e06e98.mp4

After:



https://user-images.githubusercontent.com/86483059/157105368-a7667a25-a3c2-4be5-95c6-a1ad4a743159.mp4



<!--
Add videos/screenshots of the user-facing interface in various display sizes (mainly phone, tablet, and desktop display size) to demonstrate that the changes made in this PR work correctly. Please also include videos/screenshots of the developer tools browser console, so that we can be sure that there are no console errors.
If the PR changes the UI in any of the files listed in rtl_css.py (i.e, those that have a separate .rtl.css file for styling), make sure to add screenshots with the site language set to Arabic as well.
If the changes in your PRs are autogenerated via a script and you cannot provide proof for the changes then please leave a comment "No proof of changes needed because {{Reason}}".
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Oppiabot can assign anyone for review/help if you leave a comment like the following: "{{Question/comment}} @{{reviewer_username}} PTAL"
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-your-build-fails).

ShivamJhaa committed:
Checks

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
update

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge branch 'develop' of https://github.com/oppia/oppia into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
file-1

ShivamJhaa committed:
nits

ShivamJhaa committed:
Fixed scroll issue

ShivamJhaa committed:
removed unwanted changes

ShivamJhaa committed:
fixed frontend tests

ShivamJhaa committed:
may fix e2e

ShivamJhaa committed:
maybe a fix

ShivamJhaa committed:
Undo changes from base componenet

ShivamJhaa commented:
@seanlip, @nithusha21 PTAL, thanks!

seanlip reviewed (changes requested):
LGTM, thanks for fixing this!

One question though. I think this issue also arises in the navbar. Could it be fixed there too? (In general, shouldn't we be applying the fix to a general router component rather than the navbar/footer?)

/cc @ashutoshc8101 (in case he has advice)

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into route

ShivamJhaa committed:
trying a fix as per comments

ShivamJhaa commented:
> LGTM, thanks for fixing this!
> 
> One question though. I think this issue also arises in the navbar. Could it be fixed there too? (In general, shouldn't we be applying the fix to a general router component rather than the navbar/footer?)
> 
> /cc @ashutoshc8101 (in case he has advice)

Now it's solved for both cases (navbar and footer). My apologies for the late reply as I was traveling for the past few days. PTAL, thanks.

ShivamJhaa committed:
fixed frontend coverage

seanlip commented on core/templates/base-components/base-content.component.spec.ts:
```diff
@@ -61,6 +62,13 @@ describe('Base Content Component', () => {
   }
 
+  class MockRouteService {
+    public events: Observable<NavigationEnd> = (
```
This looks quite complex. Can you add some comments explaining it in detail so that new readers know what's going on?

ShivamJhaa replied:
Done.

seanlip reviewed (changes requested):
Thanks @ShivamJhaa! No particular concerns about the code (just one small comment)  but could you please update the video to show the effect of clicking other links (e.g. in the navbar)?

ShivamJhaa committed:
added comment

ShivamJhaa commented:
@seanlip PTAL updated the video as well. Thanks!

seanlip reviewed (changes requested):
Thanks! Looks good @ShivamJhaa.

Except for one question ... if you look at the video, clicking on the navbar goes to the partnerships page, and then the page disappears and then shows up again. This seems different from the footer transitions. Why is that, and can it be avoided? (Asking because it suggests that the navbar/footer transitions aren't happening the same way.)

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into route

ShivamJhaa committed:
fixed reloading issue

ShivamJhaa commented:
> Thanks! Looks good @ShivamJhaa.
> 
> Except for one question ... if you look at the video, clicking on the navbar goes to the partnerships page, and then the page disappears and then shows up again. This seems different from the footer transitions. Why is that, and can it be avoided? (Asking because it suggests that the navbar/footer transitions aren't happening the same way.)

Fixed that now, and updated the video as well. PTAL, thanks!
The reason for reloading was navigateToPage function.

ShivamJhaa committed:
fixed reloading of foundation page

seanlip reviewed (approved):
Thanks, @ShivamJhaa! LGTM.

@ashutoshc8101 @srijanreddy98 PTAL, thanks!
